bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,1501 @@
+     * 
+     * @return security
+     */
+    public @Nullable Security getSecurity() {
+        return security;
+    }
+
+    /**
+     * Gets the Device Version (2 or 3)
+     * 
+     * @return version
+     */
+    public int getVersion() {
+        return version;
+    }
+
+    /**
+     * Set the device version
+     * 
+     * @param version device version
+     */
+    public void setVersion(int version) {
+        this.version = version;
+    }
+
+    private static final StringType OPERATIONAL_MODE_OFF = new StringType("OFF");
+    private static final StringType OPERATIONAL_MODE_AUTO = new StringType("AUTO");
+    private static final StringType OPERATIONAL_MODE_COOL = new StringType("COOL");
+    private static final StringType OPERATIONAL_MODE_DRY = new StringType("DRY");
+    private static final StringType OPERATIONAL_MODE_HEAT = new StringType("HEAT");
+    private static final StringType OPERATIONAL_MODE_FAN_ONLY = new StringType("FAN_ONLY");
+
+    private static final StringType FAN_SPEED_OFF = new StringType("OFF");
+    private static final StringType FAN_SPEED_SILENT = new StringType("SILENT");
+    private static final StringType FAN_SPEED_LOW = new StringType("LOW");
+    private static final StringType FAN_SPEED_MEDIUM = new StringType("MEDIUM");
+    private static final StringType FAN_SPEED_HIGH = new StringType("HIGH");
+    private static final StringType FAN_SPEED_FULL = new StringType("FULL");
+    private static final StringType FAN_SPEED_AUTO = new StringType("AUTO");
+
+    private static final StringType SWING_MODE_OFF = new StringType("OFF");
+    private static final StringType SWING_MODE_VERTICAL = new StringType("VERTICAL");
+    private static final StringType SWING_MODE_HORIZONTAL = new StringType("HORIZONTAL");
+    private static final StringType SWING_MODE_BOTH = new StringType("BOTH");
+    private Clouds clouds;
+
+    private ConnectionManager connectionManager;
+
+    private final SystemOfUnits systemOfUnits;
+
+    private final HttpClient httpClient;
+
+    /**
+     * Set to false when Set Command recieved to speed response
+     */
+    public boolean doPoll = true;
+
+    /**
+     * True allows one short retry after connection problem
+     */
+    public boolean retry = true;
+
+    /**
+     * Suppresses the connection message if was online before
+     */
+    public boolean connectionMessage = true;
+
+    private ConnectionManager getConnectionManager() {
+        return connectionManager;
+    }
+
+    private Response getLastResponse() {
+        return getConnectionManager().getLastResponse();
+    }
+
+    /**
+     * Initial creation of the Midea AC Handler
+     * 
+     * @param thing thing name
+     * @param ipv4Address IP address of the device
+     * @param unitProvider OH core unit provider
+     * @param httpClient http Client
+     * @param clouds cloud
+     */
+    public MideaACHandler(Thing thing, String ipv4Address, UnitProvider unitProvider, HttpClient httpClient,
```

Comment:
```suggestion
    public MideaACHandler(Thing thing, UnitProvider unitProvider, HttpClient httpClient,
```

this parameter does not seem to be used

Reply:
Done

Reply:
Fixed, only the javadoc needs to be adjusted.